neuland/simulation/generator/generator.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+#include <Math/GenVector/Cartesian3D.h>
+#include <Math/GenVector/Polar3D.h>
+#include <Math/GenVector/PxPyPzE4D.h>
+#include <algorithm>
+#include <cmath>
+#include <fmt/format.h>
+#include <TRandom.h>
+
+
+template <typename ADist, typename EDist>
+class TrackGenerator
+{
+  public:
+    TrackGenerator(const ADist& AngleDist,const EDist& EnergyDist)
+        : AngleDist_{ AngleDist }
+        , EnergyDist_{ EnergyDist }
+    {
+    }
+    void set_detector_width(double width);
+    void set_detector_length(double length);
+    void set_detector_height(double height);
+
+  private:
+    double detector_width{};
+    double detector_length{};
+    double detector_height{};
+
+    double particle_mass{};
+    static constexpr double speed_of_light{ 299792458 };
+
+    ROOT::Math::Cartesian3D<double> position{};
+
+    const ADist& AngleDist_;
+    const EDist& EnergyDist_;
+    TRandom rd_engine_;
+
+    // std::random_device rd{};
+    // std::mt19937 gen{ rd() };
+
+    auto rd_num_gen_point() -> ROOT::Math::Cartesian3D<double>;
+    auto rd_num_gen_angles()  ->  ROOT::Math::Polar3D<double>;
+    auto rd_num_gen_energy() -> double;
+    auto calculate_abs_momentum(double kinetic_energy) -> double;
+    auto calculate_momentum_energy(double kinetic_energy, double Theta, double Phi) -> ROOT::Math::PxPyPzE4D<double>;
+
+    auto calculate_postion() -> void;
+};
+
+template <typename ADist, typename EDist>
+ ROOT::Math::Polar3D<double> TrackGenerator<ADist, EDist>::rd_num_gen_angles()
+{
+    ROOT::Math::Polar3D<double> angles{};
+    angles.SetPhi(rd_engine_.Uniform(0.,M_PI));
+    angles.SetTheta(AngleDist_(rd_engine_));
+    return angles;
+}
+
+
+template <typename ADist, typename EDist>
+double TrackGenerator<ADist, EDist>::rd_num_gen_energy() 
+{
+    return EnergyDist_(rd_engine_);
+}
```

Comment:
It's unnecessary to just define a function that simply calls another function. Keep things simple. If you need to call the other function, call the other function.